nui/src/main/java/org/terasology/nui/canvas/CanvasImpl.java

```diff
@@ -193,19 +205,20 @@ public void processMousePosition(Vector2i position) {
             }
         }
 
-        mouseOverRegions.stream().filter(region -> !newMouseOverRegions.contains(region)).forEach(region ->
-            region.listener.onMouseLeave());
+        if (mouseOverRegions[pointer] != null) {
+            mouseOverRegions[pointer].stream().filter(region -> !newMouseOverRegions.contains(region)).forEach(region ->
+                    region.listener.onMouseLeave());
```

Comment:
@DarkWeird recently found a spot in Terasology where the use of a `Stream` was quite expensive. Knowing that we spent quite some processing time on NUI I'm now wondering whether this place here is a potential spot for optimization? :thinking: 

(not changing it in this PR is good, just wanted to leave my thought here)

Reply:
It is expensive if this happens thousands x thousands times per tick. (Was components x entities iterator)

Reply:
In terms of performance, this method should be run once per frame for each pointer. So, this call is `pointers * mouseOverRegions * last call's mouseOverRegions` iterations every frame. It gets worse the more interactive mouse-over regions you have. So, worst average case right now is `20 * 100's * 100's` I would think. Maybe Terasology's "Advanced Game Setup" will push this though?

I've updated it to use a for-each loop anyway.